data_pipeline/utils/pipeline_config.py

```diff
@@ -25,15 +25,15 @@ class PipelineEnvironmentVariables:
 
 
 class BigQuerySourceConfig(NamedTuple):
-    project_name: str
-    sql_query: str
+    project_name: str = ''
```

Comment:
if string values are optional then it would be better to declare them as `Optional` and set to `None` rather than an empty string.
Although in this case it would be better to make the usage of `BigQuerySourceConfig` optional.
The advantage is that we then either have an instance with all of the required fields in `BigQuerySourceConfig` or we don't. It would be caught at config parsing time whether something is missing.

Reply:
The more we can validate at config parsing time the better.